native/src/main/java/io/ballerina/stdlib/crypto/PgpEncryptionGenerator.java

```diff
@@ -0,0 +1,145 @@
+
+    public PgpEncryptionGenerator(int compressionAlgorithm, int symmetricKeyAlgorithm, boolean armor,
+                             boolean withIntegrityCheck) {
+        this.compressionAlgorithm = compressionAlgorithm;
+        this.symmetricKeyAlgorithm = symmetricKeyAlgorithm;
+        this.armor = armor;
+        this.withIntegrityCheck = withIntegrityCheck;
+    }
+
+    public void encryptStream(OutputStream encryptOut, InputStream clearIn, long length, InputStream publicKeyIn)
+            throws IOException, PGPException {
+        PGPCompressedDataGenerator compressedDataGenerator =
+                new PGPCompressedDataGenerator(compressionAlgorithm);
+        PGPEncryptedDataGenerator pgpEncryptedDataGenerator = new PGPEncryptedDataGenerator(
+                // Configure the encrypted data generator
+                new JcePGPDataEncryptorBuilder(symmetricKeyAlgorithm)
+                        .setWithIntegrityPacket(withIntegrityCheck)
+                        .setSecureRandom(new SecureRandom())
+                        .setProvider(BouncyCastleProvider.PROVIDER_NAME)
+        );
+        // Add public key
+        pgpEncryptedDataGenerator.addMethod(new JcePublicKeyKeyEncryptionMethodGenerator(
+                getPublicKey(publicKeyIn)));
+        if (armor) {
+            encryptOut = new ArmoredOutputStream(encryptOut);
+        }
+        OutputStream cipherOutStream = pgpEncryptedDataGenerator.open(encryptOut, new byte[BUFFER_SIZE]);
+        copyAsLiteralData(compressedDataGenerator.open(cipherOutStream), clearIn, length);
+        compressedDataGenerator.close();
+        cipherOutStream.close();
+        encryptOut.close();
+    }
+
+    public Object encrypt(byte[] clearData, InputStream publicKeyIn) throws PGPException, IOException {
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(clearData);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        encryptStream(outputStream, inputStream, clearData.length, publicKeyIn);
+        return ValueCreator.createArrayValue(outputStream.toByteArray());
+    }
+
+    static PGPPublicKey getPublicKey(InputStream keyInputStream) throws IOException, PGPException {
+        PGPPublicKeyRingCollection pgpPublicKeyRings = new PGPPublicKeyRingCollection(
+                PGPUtil.getDecoderStream(keyInputStream), new JcaKeyFingerprintCalculator());
+        Iterator<PGPPublicKeyRing> keyRingIterator = pgpPublicKeyRings.getKeyRings();
+        while (keyRingIterator.hasNext()) {
+            PGPPublicKeyRing pgpPublicKeyRing = keyRingIterator.next();
+            Optional<PGPPublicKey> pgpPublicKey = extractPGPKeyFromRing(pgpPublicKeyRing);
+            if (pgpPublicKey.isPresent()) {
+                return pgpPublicKey.get();
+            }
+        }
+        throw new PGPException("Invalid public key");
+    }
+
+    static void copyAsLiteralData(OutputStream outputStream, InputStream in, long length)
```

Comment:
ditto

Reply:
I have addressed this in https://github.com/ballerina-platform/module-ballerina-crypto/pull/555/commits/78e3fb4fbeab147e5c59d3a51cfec372b405651c